app/controllers/api/v2/report_templates_controller.rb

```diff
@@ -107,12 +107,22 @@ def export
       api :POST, "/report_templates/:id/generate/", N_("Generate a report template")
       param :id, :identifier, :required => true
       param :input_values, Hash, :desc => N_('Hash of input values where key is the name of input, value is the value for this input')
+      param :foreground, :bool, allow_nil: true
+      description <<-DOC
+        The reports are generated by background job. It returns an url to poll for data
+        The <b>foreground</b> parameter is used for to force report to be generated in foreground.
```

Comment:
nit: `is used to force the report to be returned synchronously`. `in foreground` is the way it's implemented, not what the user sees when he does an api request.

Reply:
I have rewritten the endpoints afterall.